File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int tLen = t.length();
        int pLen = p.length();
        long pNum = Long.parseLong(p);
        int answer = 0;
        
        for (int i = 0; i < tLen - pLen + 1; i++) {
            long subNum = Long.parseLong(t.substring(i, i + pLen));
            if (subNum <= pNum) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.kt
class Solution {
    fun solution(t: String, p: String): Int {
        val tLen = t.length
        val pLen = p.length
        val pNum = p.toLong()
        var answer: Int = 0
        
        for (i in 0 .. tLen - pLen) {
            val subNum = t.substring(i, i + pLen).toLong()
            if (subNum <= pNum) answer++
        }
        
        return answer
    }
}